hawkular-alerts-engine/src/main/java/org/hawkular/alerts/engine/impl/CassAlertsServiceImpl.java

```diff
@@ -1482,14 +1480,39 @@ private void handleResolveOptions(String tenantId, String triggerId, boolean che
 
     @Override
     public void sendData(Data data) throws Exception {
-        alertsEngine.sendData(data);
+        sendData(Collections.singleton(data));
     }
 
     @Override
     public void sendData(Collection<Data> data) throws Exception {
+        checkDataDrivenGroupTriggers(data);
         alertsEngine.sendData(data);
     }
 
+    private void checkDataDrivenGroupTriggers(Collection<Data> data) throws Exception {
+        if (!dataDrivenGroupCacheManager.isCacheActive()) {
+            return;
+        }
+        for (Data d : data) {
+            if (isEmpty(d.getSource())) {
```

Comment:
It's unlikely that a caller would provide the private SOURCE_NONE value but if they did it could cause an issue because we don't really want member triggers for the SOURCE_NONE source.  I added protection for this not here but in the needSourceMember() method, inside the cache code.